### ISSUE TITLE
Documentation and behavior of "user controlled" Properties

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/my-solar-system#251]
This is one I'll need to discuss with @AgustinVallejo, and came up as I'm starting to add `phetioDocumentation` to less-obvious PhET-iO elements.

In https://github.com/phetsims/my-solar-system/commit/0014609b, @AgustinVallejo added the following documentation to Body:

```ts
  // User modified Properties
  // Determines if the user has changes anything about the body
  public readonly userControlledProperty: Property<boolean>;

  // Determines if the user has changed the position, velocity, or mass of the body
  public readonly userControlledPositionProperty: Property<boolean>;
  public readonly userControlledVelocityProperty: Property<boolean>;
  public readonly userControlledMassProperty: Property<boolean>;
```

`userControlledProperty` does appear to match its documented behavior. It remains true after the user has made any change to a Body, and only return to false when the Reset All button is pressed.  But I've seen `userControlledProperty` used as a name in other sims, and I'm not sure that the behavior here matches the behavior in other sims. So we may need to rename this for consistency. 

The other 3 Properties do not match their documented behavior -- they are only true while the user is changing a Body, either by dragging or changing values in the Values panel.  As soon as the user is done interacting, they return to false.  This matches the behavior of "user controlled" that I've come to expect in other sims.   So we'll probably just need to adjust the doc here.

**Comment by pixelzoom** [comment]
@AgustinVallejo FYI.  In the above commits, I renamed the following Body Properties and added `phetioDocumentation`:

* userControlledMassProperty => userIsControllingMassProperty
* userControlledPositionProperty => userIsControllingPositionProperty
* userControlledVelocityProperty => userIsControllingVelocityProperty

The use of present tense in the names makes it clear that the user is controlling the associated Property now, not at some time in the past.

**Comment by pixelzoom** [comment]
In the above commits, I removed `userControlledProperty` from Body.  It was documented incorrectly there, and is in fact a reference to SolarSystemCommonModel `userControlledProperty`.  The only place that needed `body. userControlledProperty` was BodyNode, and it's just as easy to pass it `model. userControlledProperty`.

So no I need to understand SolarSystemCommonModel `userControlledProperty`, which is currently documented as:

```ts
  // Indicates whether the user has interacted with the bodies of the simulation (changing position, mass or velocity)
  public readonly userControlledProperty: Property<boolean>;
```

But searching for `userControlledProperty.value =`, I see no cases where it is set as the result of changing position, mass, or velocity.  

And to clarify... All of this is necessary because these Properties need to be described in Studio via `phetioDocumentation`.

**Comment by pixelzoom** [comment]
> But searching for `userControlledProperty.value =`, I see no cases where it is set as the result of changing position, mass, or velocity.

Correction... I found this in 

```ts
      Multilink.lazyMultilink(
        [ body.userIsControllingPositionProperty, body.userIsControllingVelocityProperty, body.userIsControllingMassProperty ],
        ( userIsControllingPosition: boolean, userIsControllingVelocity: boolean, userIsControllingMass: boolean ) => {
...
          this.userInteractingEmitter.emit();
          this.userControlledProperty.value = true;
        } );
```

This seems buggy. It does not check the Property values, so is calling `userInteractingEmitter.emit` twice -- when the user has started interacting, and when the user has finished interacting.

**Comment by AgustinVallejo** [comment]
Regarding that last examples, it's to set the cueing arrows to turn off visibility. But perhaps there are more elegant ways to achieve this, such as with the emitter. I'm talking about `BodyNode.ts:246`:
```ts
const cueingVisibleProperty = new DerivedProperty( [ userControlledProperty ], wasDragged => ( options.draggable && !wasDragged ) );
```
Also to show the Center of Mass button in MSS, which appears after user interactions (except playing, that's why the playing->userControlled only shows up in Kepler. `MySolarSystemScreenView.ts:176`:
```ts
this.followCenterOfMassButton = new TextPushButton( MySolarSystemStrings.followCenterOfMassStringProperty, {
  visibleProperty: model.userControlledProperty,
```

**Comment by pixelzoom** [comment]
In a meeting with @AgustinVallejo and @arouinfar, we concluded that:

* The purpose of `userControlledProperty` is unclear, and overloaded.
* It does not have the sematics that `userControlledProperty` has in other sims, and should be renamed. We compromised on `userHasInteractedProperty`.
* It should be `phetioReadOnly: true` and `phetioDocumentation: 'For internal use only'`.

@AgustinVallejo addressed this in the above commits.
